### PR TITLE
Use cached variables for candlepin and foreman dump check

### LINKED ISSUE #314: clone_foreman_dump and clone_candlepin_dump are not cached
This errors out when using --start-at-task

### PR DESCRIPTION
**Pull request by johnpmitsch**
Fixes #314

### CODE REVIEW

**Comment by johnpmitsch** [comment]
yee-haw

File: roles/satellite-clone/tasks/main.yml [hunk 1]
```diff
@@ -140,7 +140,7 @@
       The postgres user does not have access to the files in {{ backup_dir }}.
       Please move the backup directory to a different directory with the correct
       permissions. Avoid using /root.
-  when: foreman_dump and candlepin_dump and (clone_access_foreman_dump or clone_access_candlepin_dump)
+  when: clone_foreman_dump_exists and clone_candlepin_dump_exists and (clone_access_foreman_dump or clone_access_candlepin_dump)
```
Comment (sthirugn) [inline roles/satellite-clone/tasks/main.yml]: `clone_access_foreman_dump` and `clone_access_candlepin_dump` can be renamed to `clone_no_access_foreman_dump` and `clone_no_access_candlepin_dump` to make more sense of these variables, I had to read it twice to understand it :)

**Comment by sthirugn** [comment]
Minor comment, but not a blocker for this PR.

ACK

**Comment by sthirugn** [comment]
ACK

**Comment by johnpmitsch** [comment]
yee-haw

**Pull request merged**